### PR TITLE
Better error handling on blueflood side.

### PR DESCRIPTION
**Pull request by jeeyoungk**
* all the errors are instances of BackendError.
* added tests for common error cases (timeout, etc).

### CODE REVIEW

**Comment by jeeyoungk** [comment]
CC: @Nathan-Fenner @achow

File: api/backend/blueflood/blueflood.go [hunk 1]
```diff
@@ -66,6 +67,13 @@ const (
 type BluefloodClientConfig struct {
 	BaseUrl  string
 	TenantId string
+	Timeout  time.Duration
+}
+
+type sampler struct {
```
Comment (jeeyoungk) [inline api/backend/blueflood/blueflood.go]: pullin' it out.

**Comment by Nathan-Fenner** [comment]
LGTM

**Comment by achow** [comment]
LGTM, can I merge #92 before this? :)

**Pull request merged**